### ISSUE TITLE
Links on reverse strand have wrong coordinates

### ISSUE THREAD

**Issue by iimog** [AliTVTeam/AliTV-perl-interface#52]
It seems like links that are twisted have the wrong coordinates as if start and stop coordinates of those hits are relative to the reverse strand (so 0 coordinate at the end).
Links on the forward strand seem to be ok.

Look at the following image for clarification:
![alitv](https://cloud.githubusercontent.com/assets/7403236/16268686/996b80ac-388f-11e6-86c4-3cb262bd1f35.png)
Look at the links from the turquise "ycf2" gene. It should match while only being twisted. But the link from top to bottom hits as far from the end as it should hit from the start (notice the 0 coordinate as the sequence is shifted). The same applies for the link from bottom to top. It should hit from 19k to 25k instead it hits on 30k to 24k on the 49k long sequence (49-19=30 and 49-25=24).

**Comment by greatfireball** [comment]
Hey @iimog 

thanks for opening that bug report... It is under investigation.

**Comment by greatfireball** [comment]
Created simulated set of sequences to identify the source of the bug.

The simulator is available here (https://gist.github.com/greatfireball/171807f737c5a42ac62ff50ce09be159)

**Comment by greatfireball** [comment]
To reproduce the error I used the attached file set. Which was generated with the `simulate.pl` script:

```
Initialized random generator using 628723252
Percentage A/C/G/T: 25/25/25/25 %
Motif will be at position 4455 with a length of 1500 bp
0...250...500...750...1000...1250...1500...1750...2000...2250...2500...2750...3000...3250...3500...3750...4000...4250...4500...4750...5000...5250...5500...5750...6000...6250...6500...6750...7000...7250...7500...7750...8000...8250...8500...8750...9000...9250...9500...9750...

Extracting sequence parts...
Total length of parts is 10000 (4455+1500+4045)
```

[simulate.pl-result-srand-628723252.fasta.zip](https://github.com/AliTVTeam/AliTV-perl-interface/files/333944/simulate.pl-result-srand-628723252.fasta.zip)

**Comment by greatfireball** [comment]
After splitting of the input file into three fasta files, I ran `alitv.pl *.fasta` and produced the following image.
![wrong_alitv_visualization](https://cloud.githubusercontent.com/assets/7020976/16366453/eabf0556-3c17-11e6-8a7b-d2478eaf9df6.png)

Due to, the middle sequence is reverse complement to the other sequences, a twist of the link is expected. Nevertheless, only a single link between the sequences is expected as is the case between the upper and the bottom sequence. Therefore, the wrong number of links are generated.

This error can be reproduced with other sets.

**Comment by greatfireball** [comment]
The error is caused by Bioperl itself. It seems that Bioperl handles the reverse complement positions in MAF files wrong. This is in agreement to an issue bioperl/bioperl-live#147 reported on May, 3rd 2016.

This causes the creation of to links on each sequence, due to the existence of a link is based on its coordinates.

**Comment by cjfields** [comment]
@greatfireball the BioPerl issue with MAF data has been fixed using the patch from @dandeman.

**Comment by greatfireball** [comment]
Thx4information @cjfields 

This will require bioperl 1.7 or above. There was an integreation in 1.6 tree, but no extra version number, so the workaround is required for bioperl version up to 1.006925

I need to adjust the check if the workaround is required accordingly.

**Comment by cjfields** [comment]
@greatfireball yeah, I pulled that in for the Ensembl folks, who are targeting that branch. 

I'm planning on pushing another 1.7 release candidate out today with a full release hopefully by this weekend.

**Comment by greatfireball** [comment]
Fixed by commit e3f9c7c15d2844ba62208d071ce3060b77ca3125
